addons/web/static/src/search/cog_menu/cog_menu.js

```diff
@@ -18,59 +18,62 @@ const cogMenuRegistry = registry.category("cogMenu");
+            Dropdown,
+        };
+        static props = {
+            ...T.props,
+            getActiveIds: { type: T.props.getActiveIds, optional: true },
+            context: { type: T.props.context, optional: true },
+            resModel: { type: T.props.resModel, optional: true },
+            items: { ...T.props.items, optional: true },
+        };
+        static defaultProps = {
+            ...T.defaultProps,
+            items: {},
+        };
 
-    setup() {
-        super.setup();
-        onWillStart(async () => {
-            this.registryItems = await this._registryItems();
-        });
-        onWillUpdateProps(async () => {
-            this.registryItems = await this._registryItems();
-        });
-    }
+        setup() {
+            super.setup();
+            onWillStart(async () => {
+                this.registryItems = await this._registryItems();
+            });
+            onWillUpdateProps(async () => {
+                this.registryItems = await this._registryItems();
+            });
+        }
 
-    get hasItems() {
-        return this.cogItems.length || this.printItems.length;
-    }
+        get hasItems() {
+            return this.cogItems.length || this.printItems.length;
+        }
 
-    async _registryItems() {
-        const items = [];
-        for (const item of cogMenuRegistry.getAll()) {
-            if ("isDisplayed" in item ? await item.isDisplayed(this.env) : true) {
-                items.push({
-                    Component: item.Component,
-                    groupNumber: item.groupNumber,
-                    key: item.Component.name,
-                });
+        async _registryItems() {
+            const items = [];
+            for (const item of cogMenuRegistry.getAll()) {
+                if ("isDisplayed" in item ? await item.isDisplayed(this.env) : true) {
+                    items.push({
+                        Component: item.Component,
+                        groupNumber: item.groupNumber,
+                        key: item.Component.name,
+                    });
+                }
             }
+            return items;
         }
-        return items;
-    }
 
-    get cogItems() {
-        return [...this.actionItems, ...this.registryItems].sort((item1, item2) => {
-            const grp = (item1.groupNumber || 0) - (item2.groupNumber || 0);
-            if (grp !== 0) {
-                return grp;
-            }
-            return (item1.sequence || 0) - (item2.sequence || 0);
-        });
-    }
-}
+        get cogItems() {
+            return [...this.actionItems, ...this.registryItems].sort((item1, item2) => {
+                const grp = (item1.groupNumber || 0) - (item2.groupNumber || 0);
+                if (grp !== 0) {
+                    return grp;
+                }
+                return (item1.sequence || 0) - (item2.sequence || 0);
+            });
+        }
+    };
+
+export class CogMenu extends CogMenuMixin(ActionMenus) {}
```

Comment:
Good job to have been able to make that work. Most people don't really know about js mixins, and we don't use them a lot in Odoo. For that reason though, I'm not a big fan of this. It would be nice if we could find an alternative architecture

Reply:
An alternative would be to add an optional props `getExtraPrintItems` (or something like that) to ActionMenus, which is an async callback. In web, this props isn't given. It is called in the `beforeOpen` callback you'll provide to the Dropdown (as suggested in my other comment). In account, in your jsClass, you define the props and give it to the ActionMenus/CogMenu. In that callback, you can do your call to `get_extra_print_items`. As far as I understand, it should do the trick, and you could get rid of the mixin.